### ISSUE TITLE
Middleware regex is bad and you should feel bad

### ISSUE THREAD

**Issue by jasisk** [krakenjs/kraken-devtools#24]
[This guy](https://github.com/krakenjs/kraken-devtools/blob/77742838356a6d2e81ff85ae8699ed93b39104dc/lib/middleware.js#L45).

The problem is that `path.extname` will return an extension pretty optimistically.
```sh
$ node -e "require('path').extname('/route.ohmy/') === '' || process.exit(1)" || echo 'uh oh'
uh oh
```

So, a route like `/route.ohmy/` returns `.ohmy` for `ext`. The subsequent regex checks that extension [assuming it's at the end of the path](https://github.com/krakenjs/kraken-devtools/blob/77742838356a6d2e81ff85ae8699ed93b39104dc/lib/middleware.js#L46). Since, in this case, it's not at the end of the path, the `regex.exec` fails (returning null), throwing when you [try to get `[1]` on it](https://github.com/krakenjs/kraken-devtools/blob/77742838356a6d2e81ff85ae8699ed93b39104dc/lib/middleware.js#L53).

**Comment by pvenkatakrishnan** [comment]
hey @jasisk let me know if i can jump into help here.

**Comment by jasisk** [comment]
I swear I had a fix for this forever ago but it's lost (keeping it local + computer spring cleaning likely blew it away and I don't have my backup handy). Will take a look.

**Comment by olivierIllogika** [comment]
Similar issue, I have urls with '.' and '+' which of course explode at the regex call.

This line change fixes it for me but is in no way a proper solution to the core problem:

            ext = path.extname(req.path).replace(/\./g, '\\.').replace(/\+/g, '\\+');

**Comment by adurrive** [comment]
It's not addressing the optimistic extensions (which are pretty much everywhere) but I've submitted a fix which takes into consideration the behaviour of path.extname: https://github.com/krakenjs/kraken-devtools/pull/50

**Comment by jasisk** [comment]
Fixed by #38.
